### PR TITLE
Add setup.py for system install and pip

### PR DESCRIPTION
**Pull request by kmicc**
Spitted up from #187, implements #180.

### CODE REVIEW

**Comment by oroulet** [comment]
If you nobody has comments on this one and the edit button, please merge it so we remove number of PR

**Comment by kmicc** [comment]
I'm not very familiar with pip, so I have some questions about pip:
- Which version of QTT do I get when using pip?
- Does pip use this github repository here to install QTT or is their another place where it gets it from?
- How does pip know about the existence of QTT?

And for testing purposes:
- Can I install QTT via pip without destroying my existing QTT installation (deb-package)?
- How can I remove the QTT-pip-installation?

**Comment by oroulet** [comment]
If you use python, you should use pip! https://packaging.python.org/en/latest/current/

sudo pip3 install qtodotxt --pre
sudo pip3 uninstall qtodotxt

the uploaded version is the one I uploaded to pip, so my branch. just test it works so remove it. so I will upload master and the released version when a new version is released

**Comment by kmicc** [comment]
Ok, I will test it.

Some more questions:
* Is that setup.py needed by pip? 
* Has it a use independent from pip?
I think the answers to that questions should be documented either in the source of setup.py code and/or at an appropriate place somewhere in the docs.

Can you also provide a link to the places where the package is uploaded to and where it can be administrated? Thanks.

**Comment by oroulet** [comment]
setup.py is the standard installation file for any python program (like Makefile for C/C++ but part of python). I believ 99% of the existing python programs have it. documenting it is almost the same as documenting the for loop of python.
using setup.py you can install program to a unix machine, create rpm, create windows installer, upload to pip, etc... all with one simple command
I know the file is about correct, but it would be nice if you could look at the different values and comment if you have opinions about it (like authors, name, ..) if not we can merge it and people using that system can complain later and we can fix it. It is a self contained file that cannot break the rest of the code

**Comment by kmicc** [comment]
Thanks for your explaination. I see I still you to learn many things about python.

About the values: Seems to be mostly ok.
* "Programming Language :: Python :: 3", "Programming Language :: Python :: 2": Are we really supporting python2? If yes, we should add tests for python2 to the ever failing travis tests.
* version="1.5beta": Well, we have no 1.5beta release (and even didn't announced a new 1.5 release), so I'm not too happy with this version labeling. Maybe "master", "devel", "current-head" or something similiar, i.e. something without a concrete version number, would be more appropriate.

**Comment by kmicc** [comment]
Just tested pip install on a completely fresh installed system. 

Installation was successful, but got a warning during installation: 
> package init file 'qtodotxt/ui/resources/css/__init__.py' not found (or not a regular file)

Running the application failed:
> ImportError: No module named 'PySide'

File: README.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ QTodoTxt is a cross-platform UI client for todo.txt files (see [todo.txt](http:/
 
 Latest stable release (v1.4.0):
 
+- Everywhere: pip install qtodotxt
```
Comment (kmicc) [inline README.md]: should it not be "sudo pip3 ..." as written in the comments?

**Comment by oroulet** [comment]
pip is the name of the command sometimes pip3 on some distributions that support both python2 and python3. so better have only pip in documentation

> Are we really supporting python2?
No idea. I can remove that line

> version="1.5beta": Well, we have no 1.5beta release 
As far as I know the last release is 1.4. to upload something to pip it must have a version number. If a string is added to a version number the package is classified as pre-release. I can use 1.5alpha1 if you want. then it will be 1.5alpha2 then 1.5alpha3 until we are happy and release 1.5, thus make the package available as a release

>> ImportError: No module named 'PySide'
I did not specify pyside dependency in setup.py. I do not know what is the best practice here but I know that specifying it would have triggered the download and compilation of both qt and pyside on your pc. It is usually a better idea to install qt and pyside through your distribution, not specifiying it help to avoid unwanted compilation of qt....

**Comment by oroulet** [comment]
> package init file 'qtodotxt/ui/resources/css/init.py' not found

We ignore that one for the time being. it is a hack to avoid to rearchitecture the application

**Comment by kmicc** [comment]
* 1.5alpha1, 1.5alpha2 etc. sounds good!
* I'm also unsure about python2, especially because bin/qtodotxt states "#!/usr/bin/env python3", but bin/qtodotxt.pyw states "#!/usr/bin/env python". Will try whether QTT runs with python2 under windows and give you the results.
* Ok, I thought pip installation were "all-inclusive", but you gave really good arguments why not add the additional dependencies. Will try pip install with installed pyside and give you the results.

**Comment by mNantern** [comment]
@kmicc Since v1.4 QTT is only compatible with python 3 (see https://github.com/QTodoTxt/QTodoTxt/issues/111)

**Comment by kmicc** [comment]
@mNantern: Thanks for the information. I can confirm it is *not* running with Python 2.7.

The pip installation tests went well: Tested on Debian 8.3 with python 3.4.2 and on Windows XP with python 3.4.4. (By the way: The offical installation method of PySide for Windows is using pip...)

Concerning the PySide dependencies: Maybe we can print out a warning when calling setup.py and/or installing QTT and/or calling QTT without installed PySide? Just an idea in the case that this can be implemented without too much effort. But this could also be done outside this PR.

One last thing: I think in README.md we should give the platform specific packages a higher precedence, i.e. inserting pip install at the forth position. The reason is that the platform specific packages create desktop icons, start menu entries and other stuff that is useful for the end user. (Patch coming)

**Pull request merged**